config/crd/bases/carto.run_clusterdeliveries.yaml

```diff
@@ -48,6 +48,21 @@ spec:
             type: object
           spec:
             properties:
+              params:
+                items:
+                  properties:
+                    name:
+                      type: string
+                    overridable:
+                      type: boolean
+                    value:
+                      x-kubernetes-preserve-unknown-fields: true
+                  required:
+                  - name
+                  - overridable
```

Comment:
do we want to make these required, or, could we make it possible to omit it?

Reply:
e.g.,

```yaml
  apiVersion: carto.run/v1alpha1
  kind: ClusterSupplyChain
  metadata:
    name: foo
  spec:
    selector:
      foo: "bar"
    resources:
      - name: source-provider
        templateRef:
          kind: ClusterSourceTemplate
          name: source-template
        params:
          - name: foo
            value: ""
```

fails as such:

```
error validating data: 
  ValidationError(ClusterSupplyChain.spec.resources[0].params[0]): 
     missing required field "overridable" in run.carto.v1alpha1.ClusterSupplyChain.spec.resources.params
```

Reply:
If we end up choosing this implementation, yes, let's have a default to false story.